src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -845,6 +845,33 @@ private void ParseProject(string firstLine)
                         line = ReadLine();
                     }
                 }
+                else if (line.StartsWith("ProjectSection(SolutionItems)", StringComparison.Ordinal))
+                {
+                    // Example section:
+                    // ProjectSection(SolutionItems) = preProject
+                    //   docs\README.md = docs\README.md
+                    // EndProjectSection
+
+                    // We have a SolutionItems section.  Each subsequent line should identify
+                    // a solution item.
+                    line = ReadLine();
+                    while ((line != null) && (!line.StartsWith("EndProjectSection", StringComparison.Ordinal)))
```

Comment:
nit:
```suggestion
                    while (((line = ReadLine()) != null) && (!line.StartsWith("EndProjectSection", StringComparison.Ordinal)))
```

Small personal preference, but some others do it differently, so you should probably reject this.

Reply:
I see the logic in this line, and I agree with it as a refactor. This is another case of copying the existing patterns to lower reviewer overhead, etc, that's all.